build/config/pom.xml

```diff
@@ -121,6 +126,125 @@
+                                    Rules can start with a `-Rinternal ` or a `-Rpublic ` prefixes.
+                                    This would mean that a corresponding pattern will only be applied to filtering of internal/public rules.
+                                    If no prefix is specified then rule is considered to be both public and internal at the same time.
+                                -->
+                                <argument>^org\.hibernate\.query\.Query$</argument>
+                                <argument>^org\.hibernate\.query\.SelectionQuery$</argument>
+                                <argument>^org\.hibernate\.query\.MutationQuery$</argument>
+                                <argument>^org\.hibernate\.query\.BindableType$</argument>
+
+                                <argument>^org\.hibernate\.metamodel\.MappingMetamodel$</argument>
+                                <argument>^org\.hibernate\.metamodel\.mapping\.Bindable$</argument>
+                                <argument>^org\.hibernate\.metamodel\.mapping\.SelectableMapping$</argument>
+                                <argument>^org\.hibernate\.metamodel\.mapping\.SelectablePath$</argument>
+                                <argument>^org\.hibernate\.metamodel\.model\.domain\.JpaMetamodel$</argument>
+
+                                <argument>^org\.hibernate\.boot\.spi\.BootstrapContext$</argument>
+                                <argument>^org\.hibernate\.collection\.spi\.PersistentCollection$</argument>
+                                <argument>^org\.hibernate\.type\.spi\.TypeConfiguration$</argument>
+                                <argument>^org\.hibernate\.query\.spi\.ScrollableResultsImplementor$</argument>
+                                <argument>^org\.hibernate\.query\.sqm\.spi\.SqmCreationContext$</argument>
+
+                                <!-- Rules for things used in tests: -->
+                                <argument>-Rinternal ^org\.hibernate\.SessionEventListener$</argument>
+                                <argument>-Rinternal ^org\.hibernate\.tool\.schema\.spi\.SchemaCreator$</argument>
+                                <argument>-Rinternal ^org\.hibernate\.tool\.schema\.spi\.SchemaValidator$</argument>
+                                <argument>-Rinternal ^org\.hibernate\.tool\.schema\.spi\.SchemaMigrator$</argument>
+                                <argument>-Rinternal ^org\.hibernate\.tool\.schema\.spi\.SchemaDropper$</argument>
+                                <argument>-Rinternal ^org\.hibernate\.tool\.schema\.spi\.SchemaManagementTool$</argument>
+                                <argument>-Rinternal ^org\.hibernate\.tool\.schema\.spi\.ExtractionTool$</argument>
+                                <argument>-Rinternal ^org\.hibernate\.tool\.schema\.spi\.ExecutionOptions$</argument>
+                                <argument>-Rinternal ^org\.hibernate\.collection\.spi\.LazyInitializable$</argument>
```

Comment:
I'd remove these exceptions, see below: it's mostly fine to use internal/incubating stuff in tests.